maven-enforcer-rules/src/main/java/io/helidon/build/maven/enforcer/rules/DependencyIsValidCheck.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.build.maven.enforcer.rules;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.function.Function;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.artifact.DefaultArtifact;
+import org.apache.maven.artifact.versioning.ArtifactVersion;
+import org.apache.maven.artifact.versioning.InvalidVersionSpecificationException;
+import org.apache.maven.artifact.versioning.OverConstrainedVersionException;
+import org.apache.maven.artifact.versioning.VersionRange;
+
+/**
+ * A function that will return {@code true} if the given maven coordinate is valid.
+ */
+class DependencyIsValidCheck implements Function<Artifact, Boolean> {
+    static final String JAKARTA_RENAMED = "jakarta-renamed.properties";
+    static final String JAKARTA_VERSIONS = "jakarta-versions.properties";
+    static final String JAKARTA_GROUPS = "jakarta-groups.properties";
+
+    private static final Map<String, VersionRange> PACKAGE_TO_VERSIONS = loadVersions();
+    private static final Map<String, String> PACKAGE_TO_RENAMED = loadRenamed();
+    private static final Map<String, String> GROUP_TO_PACKAGE = loadGroups();
+
+    private final String namespace;
+    private final List<Pattern> excludedGavRegExs;
+
+    DependencyIsValidCheck(String namespace,
+                           List<Pattern> excludedGavRegExs) {
+        this.namespace = namespace;
+        this.excludedGavRegExs = excludedGavRegExs;
+    }
+
+    @Override
+    public Boolean apply(Artifact gav) {
+        String groupPackageName = toPackage(gav.getGroupId());
+        if (isExcluded(groupPackageName)) {
+            return true;
+        }
+
+        if (groupPackageName.equals("javax.servlet")
+                || groupPackageName.equals("jakarta.servlet")) {
+            return false;
+        }
+
+        try {
+            if (HelidonDependenciesRule.JAKARTA.equals(namespace)) {
```

Comment:
`equalsIgnoreCase`

Reply:
technically that is not needed here, but will add it for extra robustness.

Reply:
pushed